tests/unit-tests/test_utils.py

```diff
@@ -658,7 +631,54 @@ def test_save_embeddings_with_id_mapping(num_embs, backend):
         assert len(saved_emb) == len(embs['n2'])
         assert_equal(embs['n2'][nid_mappings['n2']].numpy(), saved_emb.numpy())
 
-def test_save_embeddings():
+def helper_save_multiple_embeddings(tmpdirname):
+    random_emb = th.rand((103, 12))
+    type0_random_emb = LazyDistTensor(random_emb, th.arange(103))
+    random_emb = th.rand((205, 12))
+    type1_random_emb = LazyDistTensor(random_emb, th.arange(205))
+
+    emb = {
+        "type0": type0_random_emb,
+        "type1": type1_random_emb
+    }
+    save_embeddings(tmpdirname, emb, 0, 4, save_embed_format='pytorch')
+    save_embeddings(tmpdirname + '_hdf5', emb, 0, 4, save_embed_format='hdf5')
+    emb = {
+        "type0": type0_random_emb,
+        "type1": type1_random_emb
+    }
+    save_embeddings(tmpdirname, emb, 1, 4)
+    save_embeddings(tmpdirname + '_hdf5', emb, 1, 4, save_embed_format='hdf5')
+    emb = {
+        "type0": type0_random_emb,
+        "type1": type1_random_emb
+    }
+    save_embeddings(tmpdirname, emb, 2, 4)
+    save_embeddings(tmpdirname + '_hdf5', emb, 2, 4, save_embed_format='hdf5')
+    emb = {
+        "type0": type0_random_emb,
+        "type1": type1_random_emb
+    }
+    save_embeddings(tmpdirname, emb, 3, 4)
+    save_embeddings(tmpdirname + '_hdf5', emb, 3, 4, save_embed_format='hdf5')
+
+    embs_shuffled = {
+        'type0': (th.arange(21), th.arange(21)),
```

Comment:
Why this is different from L658

Reply:
For shuffled node embeddings, the input is a tuple of two tensors, one for embeddings and one for node ids.